store/tikv/region_cache.go

```diff
@@ -1642,18 +1795,85 @@ const (
 	unreachable
 )
 
+func (s *Store) startHealthCheckLoopIfNeeded(c *RegionCache) {
+	// This mechanism doesn't support non-TiKV stores currently.
+	if s.storeType != TiKV {
+		logutil.BgLogger().Info("[health check] skip running health check loop for non-tikv store",
+			zap.Uint64("storeID", s.storeID), zap.String("addr", s.addr))
+		return
+	}
+
+	// It may be already started by another thread.
+	if atomic.CompareAndSwapInt32(&s.needForwarding, 0, 1) {
+		go s.checkUntilHealth(c)
+	}
+}
+
+func (s *Store) checkUntilHealth(c *RegionCache) {
+	ticker := time.NewTicker(time.Millisecond * 500)
+	lastCheckPDTime := time.Now()
+
+	// TODO(MyonKeminta): Set a more proper ctx here so that it can be interrupted immediately when the RegionCache is
+	// shutdown.
+	ctx := context.Background()
+	for {
+		select {
+		case <-c.closeCh:
+			return
+		case <-ticker.C:
+			if time.Since(lastCheckPDTime) > time.Second*30 {
+				lastCheckPDTime = time.Now()
+
+				storeMeta, err := c.PDClient().GetStore(ctx, s.storeID)
+				if err != nil {
+					logutil.BgLogger().Warn("[health check] failed to check unhealthy store's meta from pd", zap.Error(err))
+				}
+				if storeMeta == nil || storeMeta.State == metapb.StoreState_Tombstone {
+					// TODO(MyonKeminta): Should the region meta on this store be reloaded?
+					return
+				}
+			}
+			// The store is deleted due to metadata change. No need to check its health.
+			if s.getResolveState() == deleted {
```

Comment:
It's not enough. When a store is `needForwarding`, it's seldom `reResolve`d.

Reply:
Ah, there is a timer to recheck all stores at the interval of 60s by default.

Reply:
What should we do when it's not resolved yet? Should we skip until it's resolved or deleted?

Reply:
Call `reResvole` in the loop.

Reply:
I'm afraid this won't work as we expected. After resolving it creates a new `Store` object and discards the old one. I think in this case the loop will exit since the `Store`'s state becomes deleted, and another access to this store will trigger the check loop on the new `Store` object.